theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,273 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnTransformerDesc(COp):
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+    __props__ = ('precision',)
+    params_type = ParamsType(precision=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        header_dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            header_dirs += [config.dnn.include_path]
+        return header_dirs
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        lib_dirs = []
+        if config.dnn.library_path:
+            lib_dirs += [config.dnn.library_path]
+        return lib_dirs
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, precision=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+        assert cudnn.cudnnDataType_t.has_alias(precision)
+        self.precision = precision
+
+    def make_node(self, dimensions):
+        dimensions = as_tensor_variable(dimensions)
```

Comment:
After this line, check that dimension.ndim is 1 (as a vector is expected).

Also, you must check dimension dtype (integer types) and cast it (-> int64) here also, as you did in GpuDnnTransformer, because you read it in C code.